std-bits/table/src/main/java/org/enso/table/format/xlsx/Range.java

```diff
@@ -0,0 +1,255 @@
+package org.enso.table.format.xlsx;
+
+import org.apache.poi.ss.util.CellReference;
+
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Range {
+  private static final Pattern FULL_ADDRESS = Pattern.compile("^('.+'|[^'!]+)!(.+)$");
+
+  private static String[] parseFullAddress(String fullAddress) throws IllegalArgumentException {
+    if (fullAddress == null) {
+      throw new IllegalArgumentException("fullAddress cannot be NULL.");
+    }
+
+    Matcher matcher = FULL_ADDRESS.matcher(fullAddress);
+    if (!matcher.find()) {
+      throw new IllegalArgumentException("'" + fullAddress + "' is not valid format.");
+    }
+
+    return new String[] {matcher.group(1), matcher.group(2)};
+  }
+
+  private static String unescapeSheetName(String sheetName) {
+    return sheetName.replaceAll("^'(.*)'$", "$1").replaceAll("''", "'");
+  }
+
+  private static final String ADDRESS_A1 = "\\$?[A-Z]{1,3}\\$?\\d+";
+  private static final String ADDRESS_COL = "\\$?[A-Z]{1,3}";
+  private static final String ADDRESS_ROW = "\\$?\\d+";
+  private static final String ADDRESS_RC = "R(?:\\[\\d+]|\\d+)C(?:\\[\\d+]|\\d+)";
+
+  private static final Pattern RANGE_A1 =
+      Pattern.compile("^(" + ADDRESS_A1 + ")(?::(" + ADDRESS_A1 + "))?$");
+  private static final Pattern RANGE_COL =
+      Pattern.compile("^(" + ADDRESS_COL + ")(?::(" + ADDRESS_COL + "))?$");
+  private static final Pattern RANGE_ROW =
+      Pattern.compile("^(" + ADDRESS_ROW + ")(?::(" + ADDRESS_ROW + "))?$");
+  private static final Pattern RANGE_RC =
+      Pattern.compile("^(" + ADDRESS_RC + ")(?::(" + ADDRESS_RC + "))?$");
+
+  private static int[] parseRange(String range) throws IllegalArgumentException {
+    for (Pattern pattern : new Pattern[] {RANGE_A1, RANGE_COL, RANGE_ROW, RANGE_RC}) {
+      Optional<int[]> parsed =
+          parseRange(range, pattern, pattern == RANGE_RC ? Range::parseR1C1StyleAddress : Range::parseA1StyleAddress);
+
+      if (parsed.isPresent()) {
+        return parsed.get();
+      }
+    }
+
+    throw new IllegalArgumentException("Unsupported range address: " + range);
+  }
+
+  private static Optional<int[]> parseRange(
+      String range, Pattern pattern, Function<String, int[]> parser) {
+    Matcher matcher = pattern.matcher(range.toUpperCase());
+    if (!matcher.find()) {
+      return Optional.empty();
+    }
+
+    int[] tl = parser.apply(matcher.group(1));
+    if (matcher.group(2) == null) {
+      return Optional.of(new int[] {tl[0], tl[1], tl[0], tl[1]});
+    }
+
+    int[] br = parser.apply(matcher.group(2));
+    return Optional.of(
+        new int[] {
+          Math.min(tl[0], br[0]),
+          Math.min(tl[1], br[1]),
+          Math.max(tl[0], br[0]),
+          Math.max(tl[1], br[1])
+        });
+  }
+
+  private static boolean isLetter(char c) {
+    return c >= 'A' && c <= 'Z';
```

Comment:
Nope, I was also meaning to write this. But `Character.isLetter` checks for the Unicode class and will accept non-ASCII letters too. Same goes for `isDigit`:
![image](https://user-images.githubusercontent.com/1436948/167103623-395625ea-8185-4169-af74-80736d28f9fb.png)

And I think here we want only the ASCII letters and digits.
I was surprised that Java stdlib does not provide functions for the ASCII ones, only the more complex Unicode ones. But maybe that's because the check is as simple as above.

Reply:
:facepalm: 
Maybe a comment then on why we are not using stuff from stdlib?